### PR TITLE
Queue implementation based on Dynarray

### PR DESCRIPTION
**Pull request by Althiena**

### CODE REVIEW

File: PolyEngine/Core/Src/Queue.hpp [hunk 1]
```diff
@@ -1,25 +1,226 @@
...
+			for (const T& obj : list)
+				PushBack(obj);
+		}
+
+		void Enlarge() { Reserve((Capacity + 1) * 2); }
+
+		size_t Head = 0;
+		size_t Tail = 0;
+		size_t Size = 0;
+		size_t Capacity = 1;
```
Comment (MuniuDev) [inline PolyEngine/Core/Src/Queue.hpp]: Capacity should be 0 on start. No need to allocate memory if queue won't be used (for some reason). This is working better when someone is calling Reserve just after calling empty constructor.

File: PolyEngine/UnitTests/Src/QueueTests.cpp [hunk 2]
```diff
@@ -9,47 +9,48 @@ TEST_CASE("Queue tests", "[Queue]")
 	Queue<int> q;
 	Queue<int> r(q);
 
-	REQUIRE(q.Empty() == true);
-	REQUIRE(r.Empty() == true);
+	REQUIRE(q.IsEmpty() == true);
```
Comment (MuniuDev) [inline PolyEngine/UnitTests/Src/QueueTests.cpp]: It would be very nice to have more elaborate unit tests that check for example various constructors and assign operators, resizes and reserves.

File: PolyEngine/Core/Src/Queue.hpp [hunk 3]
```diff
@@ -79,7 +79,7 @@ namespace Poly
 		bool operator!=(const Queue<T>& rhs) const { return !(*this == rhs); };
 
 		//------------------------------------------------------------------------------
-		bool IsEmpty() const { return GetSize() == 0; }
+		bool Empty() const { return GetSize() == 0; }
```
Comment (MuniuDev) [inline PolyEngine/Core/Src/Queue.hpp]: Why the rename? In Dynarray the method is called `IsEmpty()` let's revert this change and keep it consistent.

File: PolyEngine/Core/Src/Queue.hpp [hunk 4]
```diff
@@ -96,42 +96,42 @@ namespace Poly
 		//------------------------------------------------------------------------------
 		void PushBack(const T& obj)
 		{
+			if (Empty() || Head == GetNextIdx(Tail))
```
Comment (MuniuDev) [inline PolyEngine/Core/Src/Queue.hpp]: You can simply check if `GetSize() + 1 > GetCapacity()` here and in the other places.

File: PolyEngine/Core/Src/Queue.hpp (hunk 3 above)
Comment (Althiena) [inline PolyEngine/Core/Src/Queue.hpp]: You've written "make it empty" in last change requests, and I misunderstood you. Thought you want me to rename this function, but you meant to make default ctor empty ;P
